bioel/ontology.py

```diff
@@ -126,25 +139,105 @@ def load_obo(cls, filepath, name=None, prefix_to_keep=None, entity_type=None, ab
-            
+
             entities[curie] = ent
 
-        if not name :
-            if filepath.startswith('http'):
-                parsed_name = filepath.split('/')[-1].split('.')[0]
-                logger.warning(f"No ontology name provided.  Using name from provided URL: {parsed_name}")
+        if not name:
+            if filepath.startswith("http"):
+                parsed_name = filepath.split("/")[-1].split(".")[0]
+                logger.warning(
+                    f"No ontology name provided.  Using name from provided URL: {parsed_name}"
+                )
                 name = parsed_name
 
         return cls(entities=entities, types=types, name=name, abbrev=abbrev)
 
+        # synonyms = _obo_term_to_synonyms(ontology, filter_prefix=prefix_to_keep)
+        # definitions = _obo_term_to_definitions(ontology, filter_prefix=prefix_to_keep)
+        # for key
+        pass
+
+    @classmethod
+    def load_medic(cls, filepath, name=None, abbrev=None, api_key=""):
+        """
+        Read medic ontology
+
+        Parameters:
+        ----------------------
+            filepath: str (Pointing to the medic directory)
+            name: str (optional)
+            abbrev: str (optional)
+            api_key: str (optional)
+        """
+        entities = {}
+        types = []
+
+        logger.info(f"Reading medic from {filepath}")
+
+        # Attributes of the medic ontology
+        key_dict = [
+            "DiseaseName",
+            "DiseaseID",
+            "AltDiseaseIDs",
+            "Definition",
+            "ParentIDs",
+            "TreeNumbers",
+            "ParentTreeNumbers",
+            "Synonyms",
+            "SlimMappings",
+        ]
+
+        # Open the TSV file
+        with open(filepath, newline="") as tsvfile:
+            reader = csv.reader(tsvfile, delimiter="\t")
+
+            counter = 0  # First entity in the tsv file appears in line 29
+
+            ontology = []
+            for row in reader:
+                dict = {}
+                if counter > 28:
+                    for i, elements in enumerate(row):
+                        dict[key_dict[i]] = elements
+                    ontology.append(dict)
+                counter += 1
+
+        for element in ontology:
+            equivalant_cuis = [element["DiseaseID"]]
+            alt_ids = (
+                element["AltDiseaseIDs"].split("|") if element["AltDiseaseIDs"] else []
+            )
+            for alt_id in alt_ids:
+                if alt_id not in equivalant_cuis and alt_id[:2] != "DO":
+                    equivalant_cuis.append(alt_id)
+
+            entity = BiomedicalEntity(
+                cui=element["DiseaseID"],
+                name=element["DiseaseName"],
+                types="Disease",
+                aliases=element["Synonyms"],
+                definition=element["Definition"],
+                equivalant_cuis=equivalant_cuis,
+            )
+            entities[element["DiseaseID"]] = entity
```

Comment:
Can you ensure your code handles duplicate `cui`s. Just give a logger warning that a duplicate cui is seen and skip that value.